### PR TITLE
Update internal srsdb and legacy mappings

### PR DESCRIPTION
**Pull request by nyalldawson**
Supersedes https://github.com/qgis/QGIS/pull/43694

### CODE REVIEW

**Comment by rouault** [comment]
What is the process to update qgscoordinatereferencesystem_legacy.h ? I guess it is a generated file ?

**Comment by nyalldawson** [comment]
@rouault 
> What is the process to update qgscoordinatereferencesystem_legacy.h ? I guess it is a generated file ?

I just run this snippet, and then copy the results into the legacy file (and update the static "next" values accordingly)

```
for c in [c for c in QgsCoordinateReferenceSystem.validSrsIds() if c >= 63321 and c<= 100000]:
    crs = QgsCoordinateReferenceSystem.fromSrsId(c)
    print(f'  {{"{crs.authid()}", "{crs.srsid()},{crs.postgisSrid()}"}},')
```

It's honestly probably not needed at all... It's done so that the new CRSes added in proj get a stable srsid on the QGIS side across different QGIS installs, but we don't really use that srsid in QGIS core anymore anyway and any plugin which is using it will have much bigger CRS handling issues anyway...

**Review by nirvn — APPROVED** [review]

**Comment by agiudiceandrea** [comment]
Shuld it be backported to 3.16 branch?

**Comment by nyalldawson** [comment]
I don't think there's a pressing need to, no

File: src/core/proj/qgscoordinatereferencesystem.cpp [hunk 1]
```diff
@@ -2354,8 +2354,8 @@ int QgsCoordinateReferenceSystem::syncDatabase()
 
   PROJ_STRING_LIST authorities = proj_get_authorities_from_database( pjContext );
 
-  int nextSrsId = 63321;
-  int nextSrId = 520003321;
+  int nextSrsId = 63560;
+  int nextSrId = 520003560;
```
Comment (agiudiceandrea) [inline src/core/proj/qgscoordinatereferencesystem.cpp:2358]: Hi @nyalldawson, it seems there is an issue here.
The srsid 63560 - srid 520003560 ("GSK-2011 / Gauss-Kruger zone 4" EPSG:20904) is already present in the srs6.db updated with this same PR.
See https://lists.osgeo.org/pipermail/qgis-user/2021-December/050357.html
Probably triggered by https://github.com/OSGeo/PROJ/pull/2943

Comment (agiudiceandrea) [reply src/core/proj/qgscoordinatereferencesystem.cpp:2358]: See https://github.com/qgis/QGIS/pull/46492

**Pull request merged**